example/provision-aks-via-resource/main.tf

```diff
@@ -0,0 +1,124 @@
+provider "azurerm" {
+  subscription_id = var.subscription_id
+  features {}
+}
+
+resource "random_id" "prefix" {
+  byte_length = 2
+}
+
+resource "azurerm_resource_group" "launch" {
+  location = var.location
+  name     = "${var.namespace}-${random_id.prefix.hex}-rg"
+}
+
+resource "azurerm_virtual_network" "default" {
+  name                = "${var.namespace}-vpc"
+  location            = var.location
+  resource_group_name = azurerm_resource_group.launch.name
+
+  address_space = ["10.10.0.0/16"]
+
+  tags = var.tags
+}
+
+resource "azurerm_subnet" "private" {
+  name                 = "${var.namespace}-private"
+  resource_group_name  = azurerm_resource_group.launch.name
+  address_prefixes     = ["10.10.1.0/24"]
+  virtual_network_name = azurerm_virtual_network.default.name
+}
+
+resource "azurerm_kubernetes_cluster" "launch" {
+  name                = "${var.namespace}-${random_id.prefix.hex}-cluster"
+  location            = var.location
+  resource_group_name = azurerm_resource_group.launch.name
+  dns_prefix          = "${var.namespace}-${random_id.prefix.hex}"
+  # kubernetes_version  = var.kubernetes_version #needs to be k8s 1.25+
+  azure_policy_enabled      = true
+  workload_identity_enabled = true
+  oidc_issuer_enabled       = true
+
+  automatic_channel_upgrade         = "stable"
+  role_based_access_control_enabled = true
+  http_application_routing_enabled  = false
+
+  default_node_pool {
+    name                = "default"
+    node_count          = 3
+    vm_size             = "Standard_D4s_v3"
```

Comment:
**nit non-blocking:** maybe the example should use gpu

Reply:
For MVP we don't need that, but yes an example with GPU's is a good idea. I will add that to the backlog.